### PR TITLE
Add keyVaultNetworkAccess and keyVaultResourceId into securityProfile.azureKeyVaultKms to support key vault with private link

### PR DESCRIPTION
**Pull request by bingosummer**
### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-05-02-preview/managedClusters.json [hunk 1]
```diff
@@ -6059,6 +6059,24 @@
...
+        },
+        "keyVaultNetworkAccess": {
+          "type": "string",
+          "enum": [
+            "Public",
+            "Private"
+          ],
+          "default": "Public",
+          "x-ms-enum": {
+            "name": "KeyVaultNetworkAccess",
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-05-02-preview/managedClusters.json:6071]: Better to distinguish it from the property name. Maybe something like `KeyVaultNetworkAccessTypes`.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-05-02-preview/managedClusters.json [hunk 2]
```diff
@@ -6059,6 +6059,24 @@
...
+            "modelAsString": true
+          },
+          "title": "Network access of the key vault",
+          "description": "network access of key vault. The possible values are `Public` and `Private`. `Public` means the key vault allows public access from all networks. `Private` means the key vault disables public access and enables private link. The default value is `Public`."
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-05-02-preview/managedClusters.json:6075]: Capitalize `network access ...`?

**Review by FumingZhang — COMMENTED** [review]
LGTM

**Comment by FumingZhang** [comment]
By the way, I have notified the feature owner of currentOrchestratorVersion (the error in [Cross-Version Breaking Changes](https://github.com/Azure/azure-rest-api-specs/pull/19086/checks?check_run_id=6449944854)) to check the reason why the stable and preview API definitions are different.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-05-02-preview/managedClusters.json (hunk 1 above)
Comment (bingosummer) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-05-02-preview/managedClusters.json:6071]: Sure. What's the purpose of "x-ms-enum"?

Comment (FumingZhang) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-05-02-preview/managedClusters.json]: See [details](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#enums-with-x-ms-enum) of "x-ms-enum".

One example would be the following snippet from Python SDK, the name in x-ms-enum would be the class name.
```
class TrustedAccessRoleBindingProvisioningState(with_metaclass(CaseInsensitiveEnumMeta, str, Enum)):
    """The current provisioning state of trusted access role binding.
    """

    SUCCEEDED = "Succeeded"
    FAILED = "Failed"
    UPDATING = "Updating"
    DELETING = "Deleting"
```

**Comment by bingosummer** [comment]
@FumingZhang 
Hi Fuming, the PR is ready for the second round review.

Q: In the check "SDK azure-sdk-for-go", the error seems not related to this PR. Could you please have a check?

```
Swagger document contains two or more x-ms-enum extensions with the same name 'ResourceIdentityType' and different values: SystemAssigned,UserAssigned,None vs. SystemAssigned
```

**Comment by FumingZhang** [comment]
> @FumingZhang Hi Fuming, the PR is ready for the second round review.
> 
> Q: In the check "SDK azure-sdk-for-go", the error seems not related to this PR. Could you please have a check?
> 
> ```
> Swagger document contains two or more x-ms-enum extensions with the same name 'ResourceIdentityType' and different values: SystemAssigned,UserAssigned,None vs. SystemAssigned
> ```

Yes, that's a known issue, pending fix by GO SDK owner.

**Review by FumingZhang — APPROVED** [review]
LGTM

**Pull request merged**